doc/source/_toc.yml

```diff
@@ -394,6 +394,7 @@ parts:
         title: "Monitoring and Debugging"
         sections:
           - file: ray-observability/getting-started
+            title: Dashboard
```

Comment:
can we use Ray Dashboard instead? Not sure if people will confuse it with other dashboard since we talk about default dashboards in grafana in doc

Reply:
Actually I took a look at master and it seems like it's already using `Ray Dashboard` as the title and TOC, so this change doesn't seem needed.

Reply:
@scottsun94 I'm thinking about closing this PR. @matthewdeng reminded me that the docs search only works on `latest` and that's why it's not showing up in the search results. Does that make sense to you?

Reply:
Sure. I also forgot that we already changed the title in the master